### PR TITLE
KAH-68_Game-screen-design-fixes

### PR DESCRIPTION
**Pull request by WojciechKaluzinski**

### CODE REVIEW

File: lib/features/game/views/page_view_pages/count_down_page.dart [hunk 1]
```diff
@@ -24,7 +22,10 @@ class CountDownPage extends StatelessWidget {
             return Padding(
               padding: const EdgeInsets.symmetric(horizontal: 32),
               child: orientation == Orientation.portrait
-                  ? Column(children: getBody(orientation, gamePageState.questionNumber))
+                  ? Column(
+                      mainAxisAlignment: MainAxisAlignment.start,
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/count_down_page.dart]: delete. That's the default value

File: lib/features/game/views/page_view_pages/get_ready_page.dart [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+import 'package:kahootify/color_const.dart';
 
 class GetReadyPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Center(child: Text('Get ready!'));
+    return Center(
+        child: Text(
+      'Get ready!',
+      style: TextStyle(fontSize: 25, color: KColors.basedBlackColor),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/get_ready_page.dart]: remove ','

File: lib/features/game/views/page_view_pages/get_ready_page.dart [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+    ));
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/get_ready_page.dart]: add ','

File: lib/features/game/views/page_view_pages/question_page.dart [hunk 4]
```diff
@@ -22,12 +22,15 @@ class QuestionPage extends StatelessWidget {
               padding: const EdgeInsets.symmetric(horizontal: 32),
               child: orientation == Orientation.portrait
                   ? Column(children: [
+                      SizedBox(height: 20),
                       QuestionInfoRow(),
                       SizedBox(height: 20),
                       CircularCountdown(time: gamePageState.serverInfo.answerTimeLimit),
-                      Expanded(child: HeaderText(text: gamePageState.quizQuestion?.question ?? '')),
+                      Expanded(flex: 1, child: HeaderText(text: gamePageState.quizQuestion?.question ?? '')),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/question_page.dart]: that's default flex

File: lib/features/game/views/page_view_pages/question_page.dart [hunk 5]
```diff
@@ -84,22 +90,41 @@ class QuestionInfoRow extends StatelessWidget {
...
-              style: TextStyle(fontSize: 15, color: KColors.basedBlackColor, fontWeight: FontWeight.bold),
-            ),
-          ],
+        return FittedBox(
+          fit: BoxFit.fitWidth,
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: [
+              Text(
+                'Question: ' + gamePageState.questionNumber.toString(),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/question_page.dart]: Use string interpolation: 'Question: ${gamePageState.questionNumber}'

File: lib/features/game/views/page_view_pages/question_page.dart [hunk 6]
```diff
@@ -84,22 +90,41 @@ class QuestionInfoRow extends StatelessWidget {
...
+                style: TextStyle(fontSize: 15, color: KColors.basedBlackColor, fontWeight: FontWeight.bold),
+              ),
+              Text(
+                '  Difficulty: ' + questionDifficultyToString(gamePageState.quizQuestion?.difficulty ?? QuestionDifficulty.medium),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/question_page.dart]: remove whitespace! If you need padding use SizedBox and not spaces in string

File: lib/features/game/views/widgets.dart [hunk 7]
```diff
@@ -54,17 +54,21 @@ class AnswerButton extends StatelessWidget {
...
-        style: ElevatedButton.styleFrom(
-          primary: getButtonColor(),
-          alignment: Alignment.center,
-          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10.0)),
-          elevation: 5,
-          padding: const EdgeInsets.symmetric(horizontal: 16),
-          textStyle: TextStyle(color: KColors.basedBlackColor, fontSize: 15),
-          minimumSize: Size(140, 120),
+      child: SizedBox(
+        width: 120,
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: hardcoded size is a bad idea. Use some grid magic to make them same size but different for various screen sizes

File: lib/features/game/views/widgets.dart [hunk 8]
```diff
@@ -54,17 +54,21 @@ class AnswerButton extends StatelessWidget {
...
+          onPressed: () => context.read<GamePageBloc>().add(AnswerQuestion(buttonState.index)),
+          child: FittedBox(fit: BoxFit.fitHeight, child: Text(buttonState.answer)),
+          style: ElevatedButton.styleFrom(
+            primary: getButtonColor(),
+            alignment: Alignment.center,
+            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10.0)),
+            elevation: 5,
+            padding: const EdgeInsets.all(8),
+            textStyle: TextStyle(color: KColors.basedBlackColor, fontSize: 15),
+            minimumSize: Size(120, 100),
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: same thing

File: lib/features/game/views/widgets.dart [hunk 9]
```diff
@@ -78,22 +82,28 @@ class HeaderText extends StatelessWidget {
...
+  final int index;
 
-  const ResultListItem({required this.playerInfo});
+  const ResultListItem({required this.playerInfo, required this.index});
 
   static const _resultTextStyle = TextStyle(color: Colors.white, fontSize: 18, fontWeight: FontWeight.bold);
 
   @override
   Widget build(BuildContext context) {
+    final int formattedScore = playerInfo.score.toInt() * 100;
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: Shouldn't that math be reversed? Like `final int formattedScore = (playerInfo.score * 100).toInt();`

File: lib/features/game/views/widgets.dart [hunk 10]
```diff
@@ -104,10 +114,23 @@ class ResultListItem extends StatelessWidget {
       child: SizedBox(
         height: 50.0,
         child: Row(
-          mainAxisAlignment: MainAxisAlignment.spaceAround,
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: Do you need this? That does nothing when all of children are Expanded

File: lib/features/game/views/widgets.dart [hunk 11]
```diff
@@ -104,10 +114,23 @@ class ResultListItem extends StatelessWidget {
...
-            Expanded(child: Text(playerInfo.name, style: _resultTextStyle)),
-            Expanded(child: Text(playerInfo.score.toString(), style: _resultTextStyle)),
+            Expanded(flex: 1, child: SizedBox()),
+            Expanded(
+                flex: 3,
+                child: Text(
+                  myIndex.toString() + '. ' + playerInfo.name,
+                  style: _resultTextStyle,
+                  textAlign: TextAlign.start,
+                )),
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: ,

File: lib/features/game/views/widgets.dart [hunk 12]
```diff
@@ -104,10 +114,23 @@ class ResultListItem extends StatelessWidget {
...
+            Expanded(
+                flex: 2,
+                child: Text(
+                  formattedScore.toString(),
+                  style: _resultTextStyle,
+                  textAlign: TextAlign.center,
+                )),
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: ,

File: lib/features/game/views/widgets.dart [hunk 13]
```diff
@@ -104,10 +114,23 @@ class ResultListItem extends StatelessWidget {
...
+                  myIndex.toString() + '. ' + playerInfo.name,
+                  style: _resultTextStyle,
+                  textAlign: TextAlign.start,
+                )),
+            Expanded(
+                flex: 2,
+                child: Text(
+                  formattedScore.toString(),
+                  style: _resultTextStyle,
+                  textAlign: TextAlign.center,
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: remove ','

File: lib/features/game/views/widgets.dart [hunk 14]
```diff
@@ -104,10 +114,23 @@ class ResultListItem extends StatelessWidget {
...
           children: [
-            Expanded(child: Text(playerInfo.name, style: _resultTextStyle)),
-            Expanded(child: Text(playerInfo.score.toString(), style: _resultTextStyle)),
+            Expanded(flex: 1, child: SizedBox()),
+            Expanded(
+                flex: 3,
+                child: Text(
+                  myIndex.toString() + '. ' + playerInfo.name,
+                  style: _resultTextStyle,
+                  textAlign: TextAlign.start,
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: remove ','

File: lib/color_const.dart [hunk 15]
```diff
@@ -6,5 +6,6 @@ class KColors {
   static const Color basedYellowColor = Color(0xffffcf00);
   static const Color basedOrangeColor = Color(0xffee6123);
   static const Color basedRedColor = Color(0xfffa003f);
+  static const Color basedBlueColor = Color(0xff42bff5);
```
Comment (joachimschmidt) [inline lib/color_const.dart]: What is meaning of 'based'?

File: lib/features/game/views/page_view_pages/count_down_page.dart [hunk 16]
```diff
@@ -8,10 +8,8 @@ class CountDownPage extends StatelessWidget {
 
   List<Widget> getBody(Orientation orientation, int questionNumber) {
     return [
-      if (orientation == Orientation.portrait) SizedBox(height: 60),
-      HeaderText(text: 'WAIT FOR ' + questionNumber.toString() + ' QUESTION'),
-      SizedBox(height: 60),
-      CircularCountdown(time: 3),
+      Expanded(flex: 3, child: HeaderText(text: 'WAIT FOR ' + questionNumber.toString() + ' QUESTION')),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/count_down_page.dart]: use string interpolation `Expanded(flex: 3, child: HeaderText(text: 'WAIT FOR $questionNumber QUESTION')),`

File: lib/features/game/views/page_view_pages/count_down_page.dart [hunk 17]
```diff
@@ -8,10 +8,8 @@ class CountDownPage extends StatelessWidget {
 
   List<Widget> getBody(Orientation orientation, int questionNumber) {
     return [
-      if (orientation == Orientation.portrait) SizedBox(height: 60),
-      HeaderText(text: 'WAIT FOR ' + questionNumber.toString() + ' QUESTION'),
-      SizedBox(height: 60),
-      CircularCountdown(time: 3),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/count_down_page.dart:14]: Magic number make const out of it

File: lib/features/game/views/page_view_pages/question_page.dart [hunk 18]
```diff
@@ -22,12 +22,15 @@ class QuestionPage extends StatelessWidget {
               padding: const EdgeInsets.symmetric(horizontal: 32),
               child: orientation == Orientation.portrait
                   ? Column(children: [
+                      SizedBox(height: 20),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/question_page.dart]: If you need padding on whole column add it above column using padding. In this case padding is already there so just add appropriate amount

File: lib/features/game/views/page_view_pages/question_page.dart [hunk 19]
```diff
@@ -39,6 +42,7 @@ class QuestionPage extends StatelessWidget {
                       Expanded(
                         flex: 1,
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/question_page.dart]: default flex

File: lib/features/game/views/page_view_pages/count_down_page.dart [hunk 20]
```diff
@@ -8,10 +8,8 @@ class CountDownPage extends StatelessWidget {
...
+      Expanded(flex: 3, child: HeaderText(text: 'WAIT FOR ' + questionNumber.toString() + ' QUESTION')),
+      Expanded(flex: 1, child: CircularCountdown(time: 3)),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/count_down_page.dart]: default flex

File: lib/features/game/views/page_view_pages/question_page.dart [hunk 21]
```diff
@@ -39,6 +42,7 @@ class QuestionPage extends StatelessWidget {
...
                         child: Column(children: [
+                          SizedBox(height: 20),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/question_page.dart:31]: Same thing as above

File: lib/features/game/views/page_view_pages/question_page.dart [hunk 22]
```diff
@@ -49,10 +53,12 @@ class QuestionPage extends StatelessWidget {
                         child: Column(children: [
                           Expanded(flex: 1, child: HeaderText(text: gamePageState.quizQuestion?.question ?? '')),
```
Comment (joachimschmidt) [inline lib/features/game/views/page_view_pages/question_page.dart]: default flex

File: lib/features/game/views/widgets.dart [hunk 23]
```diff
@@ -54,17 +54,21 @@ class AnswerButton extends StatelessWidget {
...
-          textStyle: TextStyle(color: KColors.basedBlackColor, fontSize: 15),
-          minimumSize: Size(140, 120),
+      child: SizedBox(
+        width: 120,
+        height: 100,
+        child: ElevatedButton(
+          onPressed: () => context.read<GamePageBloc>().add(AnswerQuestion(buttonState.index)),
+          child: FittedBox(fit: BoxFit.fitHeight, child: Text(buttonState.answer)),
+          style: ElevatedButton.styleFrom(
+            primary: getButtonColor(),
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: rename to _buttonColor()

File: lib/features/game/views/widgets.dart [hunk 24]
```diff
@@ -78,22 +82,28 @@ class HeaderText extends StatelessWidget {
...
   @override
   Widget build(BuildContext context) {
-    return Padding(
-      padding: const EdgeInsets.fromLTRB(16, 30, 16, 0),
-      child: Text(text, style: TextStyle(fontSize: 25, color: KColors.basedBlackColor)),
+    return FittedBox(
+      fit: BoxFit.fitWidth,
+      child: Padding(
+        padding: const EdgeInsets.fromLTRB(16, 30, 16, 0),
+        child: Text(text, style: TextStyle(fontSize: 35, color: KColors.basedBlackColor)),
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: that's crazy large fontsize!!! Why would we ever need such a big font?

File: lib/features/game/views/widgets.dart [hunk 25]
```diff
@@ -104,10 +114,23 @@ class ResultListItem extends StatelessWidget {
       child: SizedBox(
         height: 50.0,
         child: Row(
-          mainAxisAlignment: MainAxisAlignment.spaceAround,
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
           children: [
-            Expanded(child: Text(playerInfo.name, style: _resultTextStyle)),
-            Expanded(child: Text(playerInfo.score.toString(), style: _resultTextStyle)),
+            Expanded(flex: 1, child: SizedBox()),
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: default flex

File: lib/features/game/views/widgets.dart [hunk 26]
```diff
@@ -104,10 +114,23 @@ class ResultListItem extends StatelessWidget {
...
+            Expanded(
+                flex: 3,
+                child: Text(
+                  myIndex.toString() + '. ' + playerInfo.name,
```
Comment (joachimschmidt) [inline lib/features/game/views/widgets.dart]: use string interpolation! `$myIndex.${playerInfo.name}`

or better split this Text to 2 first for position second for name and adjust flex accordingly

**Review by joachimschmidt — CHANGES_REQUESTED** [review]

File: lib/features/game/views/page_view_pages/get_ready_page.dart (hunk 2 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/get_ready_page.dart]: done

File: lib/features/game/views/page_view_pages/get_ready_page.dart (hunk 3 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/get_ready_page.dart]: done

File: lib/features/game/views/page_view_pages/question_page.dart (hunk 4 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/question_page.dart]: done

File: lib/features/game/views/page_view_pages/question_page.dart (hunk 5 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/question_page.dart]: done

File: lib/features/game/views/page_view_pages/question_page.dart (hunk 6 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/question_page.dart]: done

File: lib/features/game/views/widgets.dart (hunk 7 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 8 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 9 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 10 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 11 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 12 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 13 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 14 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/color_const.dart (hunk 15 above)
Comment (WojciechKaluzinski) [reply lib/color_const.dart]: done

File: lib/features/game/views/page_view_pages/count_down_page.dart (hunk 16 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/count_down_page.dart]: done

File: lib/features/game/views/page_view_pages/count_down_page.dart (hunk 17 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/count_down_page.dart:14]: done

File: lib/features/game/views/page_view_pages/question_page.dart (hunk 18 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/question_page.dart]: done

File: lib/features/game/views/page_view_pages/question_page.dart (hunk 19 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/question_page.dart]: done

File: lib/features/game/views/page_view_pages/count_down_page.dart (hunk 20 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/count_down_page.dart]: done

File: lib/features/game/views/page_view_pages/question_page.dart (hunk 21 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/question_page.dart:31]: done

File: lib/features/game/views/page_view_pages/question_page.dart (hunk 22 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/page_view_pages/question_page.dart]: done

File: lib/features/game/views/widgets.dart (hunk 23 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 24 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 25 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/features/game/views/widgets.dart (hunk 26 above)
Comment (WojciechKaluzinski) [reply lib/features/game/views/widgets.dart]: done

File: lib/color_const.dart (hunk 15 above)
Comment (joachimschmidt) [reply lib/color_const.dart]: done? What do you mean by done? I've asked you a question

File: lib/features/game/views/page_view_pages/count_down_page.dart (hunk 17 above)
Comment (joachimschmidt) [reply lib/features/game/views/page_view_pages/count_down_page.dart:14]: How is that const... and also save it in servers consts

**Review by joachimschmidt — APPROVED** [review]

**Pull request merged**